app/src/main/java/me/guillaumin/android/osmtracker/gpx/ExportAndShareTask.java

```diff
@@ -0,0 +1,168 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.net.Uri;
+import android.util.Log;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+import java.util.TreeMap;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import me.guillaumin.android.osmtracker.R;
+import me.guillaumin.android.osmtracker.exception.ExportTrackException;
+
+/**
+ * Exports the track and media (if any) to a a temporary file, then shares it with another app
+ */
+
+public class ExportAndShareTask extends ExportTrackTask {
+    private static final String TAG = ExportToTempFileTask.class.getSimpleName();
+
+    /**
+     * Maps from name that is passed in URI to another app to a real file in the cache dir.
+     */
+    static final TreeMap<String, File> sharedFiles = new TreeMap<String, File>();
```

Comment:
Perhaps just use the interface `Map` or `SortedMap` on the left-hand side (I think everywhere else I tend to do that)